Makefile

```diff
@@ -90,3 +90,53 @@ set-version:
 docs: # Build html docs
 	rm -f docs/external/*.md
 	cd docs && make html
+
+.PHONY: coverage
+coverage:
```

Comment:
why do you need the `coverage` target?

As I see it you really need 3 targets:
1. coverage-unit-test
2. coverage-unit-test-integration
3. coverage (which calls (1) and (2) and merge them)

 Also, please have coverage-unit-tests and coverage-integration use `test` and `integration` targets with parameter (for running pytest under the coverage "supervision")

Reply:
fixed.

Reply:
make coverage-unit-tests -> make test Coverage=True
make coverage-integration -> make integration Coverage=True

no need to split coverage unit test to 3 targets.

make coverage-combine remain
make full_coverage -> make coverage